app/controllers/admin/demo_nights_controller.rb

```diff
@@ -1,4 +1,11 @@
 class Admin::DemoNightsController < Admin::BaseController
+  before_action :check_active_demo_nights, only: :create
+
+  def index
+    demo_nights = DemoNight.all
+    @inactives = demo_nights.where(status: "closed")
+    @currents = demo_nights.where.not(status: "closed")
```

Comment:
Think those will still work with a collection. So we can pass `@demo_nights = DemoNight.all`. Give it a go and let me know if that's not the case.